src/platform/plugins/shared/ui_actions/public/service/ui_actions_service.ts

```diff
@@ -130,40 +150,56 @@ export class UiActionsService {
   };
 
   /**
-   * `addTriggerAction` is similar to `attachAction` as it attaches action to a
-   * trigger, but it also registers the action, if it has not been registered, yet.
+   * @deprecated
+   *
+   * Use `plugins.uiActions.addTriggerActionAsync` instead.
    */
   public readonly addTriggerAction = (triggerId: string, action: ActionDefinition<any>): void => {
     if (!this.actions.has(action.id)) this.registerAction(action);
     this.attachAction(triggerId, action.id);
   };
 
-  public readonly getAction = (id: string): Action => {
-    if (!this.actions.has(id)) {
+  /**
+   * `addTriggerAction` is similar to `attachAction` as it attaches action to a
+   * trigger, but it also registers the action, if it has not been registered, yet.
+   */
+  public readonly addTriggerActionAsync = (
```

Comment:
Is this meant to be a shortcut for registering actions / attaching them to a trigger with one call? Is it net new functionality?

Reply:
There is an existing `addTriggerAction` function. `addTriggerActionAsync` just adds an async version.